tests/test_integration/_async/test_document.py

```diff
@@ -504,6 +511,26 @@ async def test_save_updates_existing_doc(async_data_client):
     assert new_repo["_seq_no"] == elasticsearch_repo.meta.seq_no
 
 
+@pytest.mark.asyncio
+async def test_update_empty_field(async_client):
+    await Tags._index.delete(ignore_unavailable=True)
+    await Tags.init()
+    d = Tags(id="123", tags=["a", "b"])
+    await d.save(wait_for_active_shards=1)
+    await d.update(tags=[])
+    assert d.tags == []
+
+    while True:
+        try:
+            r = await Tags.search().execute()
+            d = r.hits[0]
+        except IndexError:
+            continue
+        else:
+            break
```

Comment:
What is this loop for? Do you need a refresh instead?

Reply:
Heh, I tried to find how other tests dealt with asynchronous updates and missed the refresh option. Yes, this would be handle much cleanly with a refresh, I'll update it.